### PR TITLE
fix always return status 200 in gin router

### PR DESCRIPTION
**Pull request by donus3**
# What module is effect ?
- [x] gin router

# What changes ?
Tell the gin contrext to set status code if the response is not nil

related to the #363

### CODE REVIEW

File: router/gin/endpoint.go [hunk 1]
```diff
@@ -61,6 +61,8 @@ func CustomErrorEndpointHandler(configuration *config.EndpointConfig, prxy proxy
 					c.Writer.Header().Add(k, v)
 				}
 			}
+
+			c.Status(response.Metadata.StatusCode)
```
Comment (kpacha) [inline router/gin/endpoint.go:65]: `response` could be nil

Comment (donus3) [reply router/gin/endpoint.go:65]: @kpacha the line https://github.com/donus3/krakend/blob/16ea2a68e1f8f287de3b041b52372eb9a786b297/router/gin/endpoint.go#L51 already check `if response != nil`

Comment (donus3) [reply router/gin/endpoint.go:65]: Btw, I will check `if response.Metadata != nil`

Comment (donus3) [reply router/gin/endpoint.go:65]: @kpacha pls reopen

**Pull request closed without merging**